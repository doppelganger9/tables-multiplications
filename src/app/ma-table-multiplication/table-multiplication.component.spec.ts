import { of } from 'rxjs';
import { StateService } from '../store/state.service';
import { TableMultiplicationComponent } from './table-multiplication.component';
import { TestBed } from '@angular/core/testing';

describe(`TableMultiplicationComponent`, () => {
  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [],
      providers: [StateService],
      declarations: [TableMultiplicationComponent]
    }).compileComponents();
  });

  it('should create the component', () => {
    const fixture = TestBed.createComponent(TableMultiplicationComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it.each`
    nombre
    ${1}
    ${2}
    ${3}
    ${4}
    ${5}
    ${6}
    ${7}
    ${8}
    ${9}
    ${10}
  `('should show table for $nombre', ({ nombre }) => {
    const fixture = TestBed.createComponent(TableMultiplicationComponent);
    const app = fixture.componentInstance;

    app.nombre$ = of(nombre);

    app.ngOnChanges();

    app.tableMultiplication$.subscribe((res) => {
      expect(res).toMatchSnapshot();
    });
  });
});
