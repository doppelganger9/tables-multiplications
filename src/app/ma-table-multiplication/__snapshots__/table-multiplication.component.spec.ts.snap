// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableMultiplicationComponent should show table for 1 1`] = `
[
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
]
`;

exports[`TableMultiplicationComponent should show table for 2 1`] = `
[
  2,
  4,
  6,
  8,
  10,
  12,
  14,
  16,
  18,
  20,
]
`;

exports[`TableMultiplicationComponent should show table for 3 1`] = `
[
  3,
  6,
  9,
  12,
  15,
  18,
  21,
  24,
  27,
  30,
]
`;

exports[`TableMultiplicationComponent should show table for 4 1`] = `
[
  4,
  8,
  12,
  16,
  20,
  24,
  28,
  32,
  36,
  40,
]
`;

exports[`TableMultiplicationComponent should show table for 5 1`] = `
[
  5,
  10,
  15,
  20,
  25,
  30,
  35,
  40,
  45,
  50,
]
`;

exports[`TableMultiplicationComponent should show table for 6 1`] = `
[
  6,
  12,
  18,
  24,
  30,
  36,
  42,
  48,
  54,
  60,
]
`;

exports[`TableMultiplicationComponent should show table for 7 1`] = `
[
  7,
  14,
  21,
  28,
  35,
  42,
  49,
  56,
  63,
  70,
]
`;

exports[`TableMultiplicationComponent should show table for 8 1`] = `
[
  8,
  16,
  24,
  32,
  40,
  48,
  56,
  64,
  72,
  80,
]
`;

exports[`TableMultiplicationComponent should show table for 9 1`] = `
[
  9,
  18,
  27,
  36,
  45,
  54,
  63,
  72,
  81,
  90,
]
`;

exports[`TableMultiplicationComponent should show table for 10 1`] = `
[
  10,
  20,
  30,
  40,
  50,
  60,
  70,
  80,
  90,
  100,
]
`;
